cli/admin/src/test/java/io/pravega/cli/admin/AbstractTlsAdminCommandTest.java

```diff
@@ -0,0 +1,122 @@
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (authEnabled) {
+            emulatorBuilder.passwdFile("../../config/" + SecurityConfigDefaults.AUTH_HANDLER_INPUT_FILE_NAME)
+                    .userName(SecurityConfigDefaults.AUTH_ADMIN_USERNAME)
+                    .passwd(SecurityConfigDefaults.AUTH_ADMIN_PASSWORD);
+        }
+
+        if (tlsEnabled) {
+            emulatorBuilder.certFile("../../config/" + SecurityConfigDefaults.TLS_SERVER_CERT_FILE_NAME)
+                    .keyFile("../../config/" + SecurityConfigDefaults.TLS_SERVER_PRIVATE_KEY_FILE_NAME)
+                    .jksKeyFile("../../config/" + SecurityConfigDefaults.TLS_SERVER_KEYSTORE_NAME)
+                    .jksTrustFile("../../config/" + SecurityConfigDefaults.TLS_CLIENT_TRUSTSTORE_NAME)
+                    .keyPasswordFile("../../config/" + SecurityConfigDefaults.TLS_PASSWORD_FILE_NAME);
+        }
+
+        localPravega = emulatorBuilder.build();
+
+        // The uri returned by LocalPravegaEmulator is in the form tcp://localhost:9090 (protocol + domain + port)
+        // but for the CLI we need to set the GRPC uri as localhost:9090 (domain + port). Because the protocol
+        // is decided based on whether security is enabled or not.
+
+        // Set the CLI properties.
+        STATE.set(new AdminCommandState());
+        Properties pravegaProperties = new Properties();
+        pravegaProperties.setProperty("cli.controller.rest.uri", "localhost:" + REST_SERVER_PORT.toString());
+        pravegaProperties.setProperty("cli.controller.grpc.uri", "localhost:" + CONTROLLER_PORT.toString());
+        pravegaProperties.setProperty("pravegaservice.zk.connect.uri", localPravega.getInProcPravegaCluster().getZkUrl());
+        pravegaProperties.setProperty("pravegaservice.container.count", "4");
+        pravegaProperties.setProperty("cli.security.auth.enable", Boolean.toString(authEnabled));
+        pravegaProperties.setProperty("cli.security.auth.credentials.username", SecurityConfigDefaults.AUTH_ADMIN_USERNAME);
+        pravegaProperties.setProperty("cli.security.auth.credentials.password", SecurityConfigDefaults.AUTH_ADMIN_PASSWORD);
+        pravegaProperties.setProperty("cli.security.tls.enable", Boolean.toString(tlsEnabled));
+        pravegaProperties.setProperty("cli.security.tls.trustStore.location", "../../config/" + SecurityConfigDefaults.TLS_CLIENT_TRUSTSTORE_NAME);
+
+        STATE.get().getConfigBuilder().include(pravegaProperties);
+
+        localPravega.start();
+
+        // Wait for the server to complete start-up.
+        TimeUnit.SECONDS.sleep(20);
```

Comment:
Do not do this. Find a different way to figure out when the server has started.

20 seconds may be too long in most cases and may not be sufficient if Travis is super-slow when your test executes.

Reply:
Will try to look for a better way to identify if the server has started.

Reply:
This has been removed.